docs/radio.rst

```diff
@@ -0,0 +1,129 @@
+
+    The ``length`` defines the maximum length, in bytes, of a message sent via
+    the radio. It can be up to 251 bytes long (254 - 3 bytes for S0, LENGTH
+    and S1 preamble).
+
+    The ``queue`` specifies the number of messages that can be stored on the
+    incoming message queue. If there are no spaces left on the queue for
+    incoming messages, then the incoming message is dropped.
+
+    The ``channel`` can be an integer value from 0 to 100 (inclusive) that
+    defines an arbitrary "channel" to which the radio is tuned. Messages will
+    be sent via this channel and only messages received via this channel will
+    be put onto the incoming message queue. Each step is 1MHz wide, based at
+    2400MHz.
+
+    The ``power`` is an integer value from 0 to 7 (inclusive) to indicate the
+    strength of signal used when broadcasting a message. The higher the value
+    the stronger the signal, but the more power is consumed by the device. The
+    numbering translates to positions in the following list of dBm (decibel
+    milliwatt) values: -30, -20, -16, -12, -8, -4, 0, 4.
+
+    The ``address`` is an arbitrary name, expressed as a 32-bit address, that's
+    used to filter incoming packets at the hardware level, keeping only those
+    that match the address you set. The default used by other micro:bit related
+    platforms is ``b'uBit\x00'``. It must be exactly 5 bytes in length.
+
+    The ``group`` is an 8-bit value (0-255) used with the ``address`` when
+    filtering messages. Conceptually, "address" is like a house/office
+    address and "group" is like the person at that address to which you want
+    to send you message.
+
+    The ``data_rate`` indicates the speed at which data throughput takes
+    place. Can be one of the following contants defined in the ``radio``
+    module : ``RATE_250KBIT``, ``RATE_1MBIT`` (the default)  or ``RATE_2MBIT``.
+
+    If ``config`` is not called then the defaults described above are assumed.
+    Since these are named arguments with sensible defaults, re-setting is as
+    simple as calling ``config()``. Calling with a subset of the named
+    arguments (for example, ``config(power=3)`` will cause all the other
+    settings to default their values.
+
+.. note::
+
+    None of the following send or receive methods will work until the radio is
+    turned on.
+
+.. py:function:: send_bytes(message)
+
+    Sends a message containing bytes.
+
+.. py:function:: receive_bytes()
+
+    Receive the next incoming message on the message queue. Returns ``None`` if
+    there are no pending messages. Messages are returned as bytes.
+
+.. py:function:: send(message)
+
+    Sends a message string. This is the equivalent of
+    ``send_bytes(bytes(message, 'utf8'))``.
+
+.. py:function:: receive()
+
+    Works in exactly the same way as ``receive_bytes`` but returns
+    strings. Equivalent to ``str(receive_bytes(), 'utf8')``. Will raise a
+    ``ValueError`` exception if conversion to string fails.
```

Comment:
Could you just say "returns whatever was sent"? That future proofs it for when we add new types.